memif-attach/main.go

```diff
@@ -67,6 +79,43 @@ type InitMsg struct {
 	Name    [32]byte
 }
 
+// AddRegionMsg message
+type AddRegionMsg struct {
+	Ack   ackMsg
+	Index uint16
+	Size  uint64
+}
+
+// AddRingMsg message
+type AddRingMsg struct {
+	Ack                ackMsg
+	Index              uint16
+	Region             uint16
+	Offset             uint32
+	Max_log2_ring_size uint8
+	Private_hdr_size   uint16
+}
+
+// AddRingMsg type
+type MemifRingT struct {
+	Pad1   [64]byte
+	cookie uint32
+	flags  uint16
+	head   uint16
+	Pad2   [56]byte
+	tail   uint16
+	desc   [0]MemifDescT
```

Comment:
My problem mainly comes from this field in struct. I have met a similar problem which so call "flexible array member" before (https://gcc.gnu.org/onlinedocs/gcc/Zero-Length.html). I believe this field should have size zero, however, when I tested it in Go program, I found that the field `desc[0]MemifDescT` occupied 2 bytes in the struct MemifRingT, and if declared it along, it had size 0. I definitely need an explanation here.